File: python/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        prev_nums = {} # num: index

        for i, num in enumerate(nums):
            complement = target - num

            if complement in prev_nums:
                return [prev_nums[num], i]

            prev_nums[num] = i

File: javascript/0001-two-sum.js
var twoSum = function(nums, target) {
    map = {}
    for(let i = 0; i < nums.length; i++) {
        const complement = target - nums[i]
        if(complement in map) return [map[complement], i]
        map[nums[i]] = i
    }
    return []
};
